lib/resque/durable.rb

```diff
@@ -66,10 +66,24 @@ def around_perform_manage_audit(*args)
           yield
         end
 
-        a.complete!
+        if re_enqueue_immediately?
+          a.re_enqueue_immediately!
+        else
+          a.complete!
+        end
       else
         yield
       end
+    ensure
+      @requeue_immediately = false
```

Comment:
I wish resque ran on object instances instead of classes. It may be safer to track this state on the queue audit instead of the job class to avoid adding global state.

Reply:
Ugh, that would mean exposing the audit to the perform method, which we don't do. Maybe nevermind.